src/collector/top_collector.rs

```diff
@@ -206,10 +204,11 @@ mod tests {
         top_collector.collect(5, 0.3);
         assert_eq!(
             top_collector.harvest(),
+            // Note: Individual segments are not sorted.
             vec![
                 (0.8, DocAddress::new(0, 1)),
+                (0.2, DocAddress::new(0, 3)),
```

Comment:
i am not fond of the assert now. The order is now very specific to the implementation. Can you assert on the set. For instance, by sorting the results (in the unit test.)

Reply:
Hm: that's actually harder than it looks, because floats don't implement `Ord`, `Hash`, or `Eq`.

I could maybe do it with `sort_by` and `partial_ord`, but I'm not sure that that would add to the clarity of the test. Thoughts?

Reply:
You can convert floats to https://docs.rs/ordered-float/latest/ordered_float/